decimal/decimal_test.go

```diff
@@ -0,0 +1,574 @@
+	}
+	if buf != nil {
+		t.Fatalf("buf is not nil on encoding of a string number with non-digit symbol")
+	}
+	if err.Error() != referenceErrMsg {
+		t.Fatalf("wrong error message on encoding of a string number with non-digit symbol")
+	}
+}
+
+func TestEncodeMaxNumber(t *testing.T) {
+	referenceErrMsg := "msgpack: decimal number is bigger than maximum supported number (10^38 - 1)"
+	decNum := decimal.New(1, DecimalPrecision) // // 10^DecimalPrecision
+	tuple := TupleDecimal{number: decNum}
+	_, err := msgpack.Marshal(&tuple)
+	if err == nil {
+		t.Fatalf("It is possible to encode a number unsupported by Tarantool")
+	}
+	if err.Error() != referenceErrMsg {
+		t.Fatalf("Incorrect error message on attempt to encode number unsupported by Tarantool")
+	}
+}
+
+func TestEncodeMinNumber(t *testing.T) {
+	referenceErrMsg := "msgpack: decimal number is lesser than minimum supported number (-10^38 - 1)"
+	two := decimal.NewFromInt(2)
+	decNum := decimal.New(1, DecimalPrecision).Neg().Sub(two) // -10^DecimalPrecision - 2
+	tuple := TupleDecimal{number: decNum}
+	_, err := msgpack.Marshal(&tuple)
+	if err == nil {
+		t.Fatalf("It is possible to encode a number unsupported by Tarantool")
+	}
+	if err.Error() != referenceErrMsg {
+		t.Fatalf("Incorrect error message on attempt to encode number unsupported by Tarantool")
+	}
+}
+
+func benchmarkMPEncodeDecode(b *testing.B, src decimal.Decimal, dst interface{}) {
+	b.ResetTimer()
+
+	var v TupleDecimal
+	var buf []byte
+	var err error
+	for i := 0; i < b.N; i++ {
+		tuple := TupleDecimal{number: src}
+		if buf, err = msgpack.Marshal(&tuple); err != nil {
+			b.Fatal(err)
+		}
+		if err = msgpack.Unmarshal(buf, &v); err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func BenchmarkMPEncodeDecodeDecimal(b *testing.B) {
+	for _, testcase := range benchmarkSamples {
+		b.Run(testcase.numString, func(b *testing.B) {
+			dec, err := decimal.NewFromString(testcase.numString)
+			if err != nil {
+				b.Fatal(err)
+			}
+			benchmarkMPEncodeDecode(b, dec, &dec)
+		})
+	}
+}
+
+func BenchmarkMPEncodeDecimal(b *testing.B) {
+	for _, testcase := range benchmarkSamples {
+		b.Run(testcase.numString, func(b *testing.B) {
+			decNum, err := decimal.NewFromString(testcase.numString)
+			if err != nil {
+				b.Fatal(err)
+			}
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				msgpack.Marshal(&decNum)
+			}
+		})
+	}
+}
+
+func BenchmarkMPDecodeDecimal(b *testing.B) {
+	for _, testcase := range benchmarkSamples {
+		b.Run(testcase.numString, func(b *testing.B) {
+			decNum, err := decimal.NewFromString(testcase.numString)
+			if err != nil {
+				b.Fatal(err)
+			}
+			var buf []byte
+			if buf, err = msgpack.Marshal(&decNum); err != nil {
+				b.Fatal(err)
+			}
+			b.ResetTimer()
+			var v TupleDecimal
+			for i := 0; i < b.N; i++ {
+				msgpack.Unmarshal(buf, &v)
+			}
+
+		})
+	}
+}
+
+func connectWithValidation(t *testing.T) *Connection {
+	conn, err := Connect(server, opts)
```

Comment:
It will be better to mark the function as a [test helper](https://pkg.go.dev/testing#T.Helper)
```suggestion
func connectWithValidation(t *testing.T) *Connection {
	t.Helper()
	conn, err := Connect(server, opts)
```

Reply:
Added.